tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/InstalledPackageViewModel.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.Common.Contracts;
+using DevHome.Common.Extensions;
+using Microsoft.UI.Xaml;
+using Windows.ApplicationModel;
+using Windows.System;
+
+namespace DevHome.ExtensionLibrary.ViewModels;
+public partial class InstalledExtensionViewModel : ObservableObject
+{
+    [ObservableProperty]
+    private string _displayName;
+
+    [ObservableProperty]
+    private string _packageFullName;
+
+    [ObservableProperty]
+    private bool _hasSettingsProvider;
+
+    [ObservableProperty]
+    private bool _isExtensionEnabled;
+
+    partial void OnIsExtensionEnabledChanging(bool value)
+    {
+        if (IsExtensionEnabled != value)
+        {
+            Task.Run(() =>
+            {
+                var localSettingsService = Application.Current.GetService<ILocalSettingsService>();
+                return localSettingsService.SaveSettingAsync(PackageFullName + "-ExtensionDisabled", !value);

```

Comment:
I had offline discussion about this, but PackageFullName is not safe to use for extension setting name for two reasons:
1) It will change when the package is updated, because Full Name includes package version, meaning the setting will be orphaned and the current state will always reset, losing the user's setting.
2) It will collide if there are multiple extensions within the package (either by multiple applications being present, or by there being multiple devhome extensions defined within the application), they'll all use the same setting if PackageFullName is used to define them.

This should be changed to use a Unique Extension Identifier, which is PackageFamilyName + ApplicationId + ExtensionId. That combination is guaranteed to be unique for a given extension, regardless of how many applications or extensions within an application exist, and that combination is also consistent across package versions (as long as the version update didn't change the extension id). This is mentioned above in the PluginWrapper changes.